Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -878,19 +887,20 @@ def ignore_property(name_to_check, regex_to_ignore):
         if ignore_property(name, ignore_regex):
             continue
         prop = algm_obj.getProperty(name)
-        # Parent algorithms store their workspaces in the ADS
-        # Child algorithms should store their workspaces in the property
-        # but they don't at the moment while the issues with history recording Python Child Algs
-        # is resolved: ticket #5157
+
         if _is_workspace_property(prop):
             value_str = prop.valueAsStr
             try:
                 retvals[name] = _api.AnalysisDataService[value_str]
```

Comment:
I think I've found a problem with trying the ADS first. It can be illustrated with the following script in the MantidPlot script window:

```python
# Run these 2 lines
test_ws = CreateSampleWorkspace(StoreInADS=True)
test_ws = CreateSampleWorkspace(StoreInADS=False)

# Now simulate deleting the workspace from the ADS using the delete button
DeleteWorkspace('test_ws')
Pause(0.5) # required to give the async delete time to happen.

# Now run this line. I would expect the variable to still be alive but it is not.
print (test_ws.getNumberHistograms())
```

I think the issue is that as `AnalysisDataService[value_str]` is first  `test_ws = CreateSampleWorkspace(StoreInADS=False)` then still goes through this line and a `weak_ptr` to the workspace is still returned the second time around (see https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/inc/MantidPythonInterface/kernel/DataServiceExporter.h#L158)

Ideally I would like to remove the `weak_ptr` from the c++ side and move it to be controlled from Python but I think this is outside of the scope of this change.

Reply:
Good catch, indeed, so do you think just swapping the order(value first, then check ADS) should be fine for the moment? Can you think of a counter-example to that? 
Btw, I've found out that the simple API specialisations for Load and RenameWorkspace do not respect the logging keyword, probably also the new keyword. It seems they set properties from a dictionary that is not the original kwargs, try:
```
ws=CreateSampleWorkspace()
RenameWorkspace('ws','ws2',EnableLogging=False)
```

Reply:
@gvardany I *think* swapping to order should do the trick. In the case where `StoreInADS=True` the `getProperty.value` for a workspace should be `None` so falling back to the ADS should be the correct thing to do.

I hadn't noticed the `Load`/`RenameWorkspace` behaviour - I would say this is a bug as all of the functions are intended to have the same behaviour with regards to these things. I'd be happy if that were sorted out in a new issue if you think it's too much work for now.

Reply:
Actually I think it would be fit to fix them here as well, this shouldn't be too hard, I will have a look.